c++/src/kj/async-io-test.c++

```diff
@@ -3574,5 +3576,45 @@ KJ_TEST("pump file to socket") {
   doTest(fs->getCurrent().createTemporary());
 }
 
+KJ_TEST("Calling abortRead() while tryRead() is in progress") {
+  // This is a stream that will only permit one call to tryRead() to be in progress at any one time
+  // Any tryRead calls won't ever complete
+  class NonConcurrentStream: public kj::AsyncInputStream {
+    public:
+    virtual kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
+      numConcurrentCalls.fetch_add(1);
+      co_await (kj::Promise<void>) kj::NEVER_DONE;
+      numConcurrentCalls.fetch_sub(1);
+      co_return 0;
+    }
+
+    virtual ~NonConcurrentStream() {}
+
+    std::atomic_int numConcurrentCalls = 0;
```

Comment:
This doesn't need to be atomic since all access is from the same thread.

Reply:
True. I originally used an int then got unnecessarily paranoid.